packages/amplify-prompts/src/progressbars/terminal.ts

```diff
@@ -0,0 +1,94 @@
+const ESC = '\u001b';
+
+const cursorUp = (n : number) : string => {
+  const dy = typeof n === 'number' ? n : 1;
+  return dy > 0 ? `${ESC}[${dy}A` : '';
+};
+
+const clearLine = () : string => `${ESC}[K`;
+
+/**
+ * Amplify Terminal instance (Re writable block)
+ */
+export class AmplifyTerminal {
+    private lastHeight : number;
+    private trailingEmptyLines : number;
+    private readonly stream: NodeJS.WriteStream;
+
+    constructor() {
+      this.lastHeight = 0;
+      this.trailingEmptyLines = 0;
+      this.stream = process.stdout;
+    }
+
+    /**
+     * Checks if the environment is tty
+     */
+    isTTY(): boolean {
+      return this.stream.isTTY;
+    }
+
+    /**
+     * Width of terminal
+     */
+    public get width() : number {
+      return this.stream.columns;
+    }
+
+    /**
+     * Height of last printed block
+     */
+    public getLastHeight() : number {
+      return this.lastHeight;
+    }
+
+    /**
+     * Height of terminal
+     */
+    public get height() : number {
+      return this.stream.rows;
+    }
```

Comment:
seems like these getters should all be private or not exist at all? If they are used for testing, is there some other way that the relevant behavior can be tested by inspecting mocks?